Zelda/Zelda.cpp

```diff
@@ -1,1205 +0,0 @@
-
-		for (int i = 0; i < 9 && !InputStream.eof(); ++i)
-		{
-			InputStream.getline(buffer,300);
-			rooms[i].setDesription(buffer);
-		}
-	}
-	else
-		cout << "Could not open file";
-	InputStream.close();
-}
-
-
-
-void Castle::setNumbersofRooms()
-{
-	for (int i = 0; i < 9; ++i)
-	{
-		rooms[i].setRoomNumber(i+1);
-	}
-}
-
-
-
-Castle::~Castle()
-{
-
-}
-
-
-
-//-------------------Player---------------------
-
-
-
-Player::Player(char const* player_Name , Room* room1 )
-{
-	name = new char [strlen(player_Name) + 1];
-	strcpy_s (name,strlen(player_Name) + 1, player_Name);
-
-	current_Room = room1;
-	princess_Pointer = nullptr;
-	is_Alive = true;
-	cash = 0;
-
-	for (int i = 0; i < 10; ++i)
-	{
-		bag[i] = nullptr;
-	}
-}
-
-
-
-Player::Player()
-{
-
-}
-
-
-
-Player::~Player()
-{
-	if (name)
-	{
-		delete[] name;
-	}
-}
-
-
-
-bool Player::getCurrentState()
-{
-	return is_Alive;
-}
-
-
-
-void Player::setCurrentState(bool currentState)
-{
-	is_Alive = currentState;
-}
-
-
-
-Princess* Player::getPrincess_Pointer()
-{
-	return princess_Pointer;
-}
-
-
-
-char* Player::getPlayerName()
-{
-	return name;
-}
-
-
-
-void Player::CashUpdater()
-{
-	cash = 0;
-
-	for (int i = 0; i < 10; ++i)
-	{
-		if(bag[i])
-		{
-			cash = cash + bag[i]->getWorth();
-		}
-	}
-}
-
-
-
-int Player::getCash()
-{
-	return cash;
-}
-
-
-
-bool Player::isBagEmpty()
-{
-	for (int i = 0; i < 10; ++i)
-	{
-		if(bag[i] != nullptr)
-			return false;
-	}
-
-	return true;
-}
-
-
-
-bool Player::isBagFull()
-{
-	bool bag_Full = true;
-
-	for (int i = 0; i < 10 && bag_Full; ++i)
-	{
-		if(bag[i] == nullptr)
-			bag_Full = false;
-	}
-
-	return bag_Full;
-}
-
-
-
-bool Player::Move(char const* direction , bool& exit_Castle)
-{
-	bool moveSuccessful = false;
-	int i = 0;
-
-	i = current_Room->directionIndex(direction);
-
-	if ((i == 3) && ((current_Room->getRoomNumber()) == 1))
-	{
-		exit_Castle = true;
-	}
-
-	else if((current_Room->getPaths())[i] != nullptr)
-	{
-		current_Room   = (current_Room->getPaths())[i];
-		moveSuccessful = true;
-
-		if(current_Room->getPrincessPresent())
-		{
-			HelperFunctions::color(AQUA);
-			cout << "The Princess is standing , waiting for you in the dark . ";
-			princess_Pointer= current_Room->getPrincessPresent();
-			current_Room->setPrincessPresent(nullptr);
-		}
-
-		HelperFunctions::color(RED);
-		cout << "\nYou have successfully been moved to Room : " << current_Room->getRoomNumber();
-	}
-
-	else
-	{
-		HelperFunctions::color(RED);
-		cout << "\nMOVE " << direction <<" is an invalid command.";
-	}
-
-	return moveSuccessful;
-}
-
-
-
-
-
-
-
-void Player::Pick(char const* itemName)
-{
-	bool picked = false;
-	bool itemFound = false;
-
-	for (int i = 0; i < 5; ++i)
-	{
-		if((current_Room->getItemsPresent())[i] != nullptr)
-		{
-			if(	strcmp((((current_Room->getItemsPresent())[i])->getItemName()) , itemName) == 0)
-			{
-				itemFound = true;
-
-				if (!isBagFull())
-				{
-					int j = 0;
-
-					for (; bag[j] != nullptr; ++j)
```

Comment:
There's a better way to find the next open spot in the Player's bag.  Create a member variable for the player class of currentBagCount.